### PR TITLE
fix: replace connection strings with environment variables

### PR DESCRIPTION
**Pull request by francescastocco**
## Introduction :pencil2:
The feature environment was not working as connection strings had been moved to environment variables.

## Resolution :heavy_check_mark:
Updated the bicep code so these are additional environment variables

## Miscellaneous :heavy_plus_sign:
List any additional fixes or improvements.

### CODE REVIEW

File: infrastructure/resource_group_level/modules/webapps/trade-finance-manager-api-no-calculated-variables.bicep [hunk 1]
```diff
@@ -96,3 +96,4 @@ module tfmApiWebapp 'webapp.bicep' = {
 }
 
 output defaultHostName string = tfmApiWebapp.outputs.defaultHostName
+output tfmApiAppSettings object = appSettings
```
Comment (francescastocco) [inline infrastructure/resource_group_level/modules/webapps/trade-finance-manager-api-no-calculated-variables.bicep:99]: This is a bit crude, I set the appSettings on tfm-api and then output them here, and then in `...calculated-variables.bicep` create a union with the outputted appSettings and calculated values & reset the appSettings on tfm-api

File: infrastructure/resource_group_level/main.bicep [hunk 2]
```diff
@@ -180,43 +171,50 @@ var portalApiSecureConnectionStrings = {
...
+  UKEF_TFM_API_SYSTEM_KEY: UKEF_TFM_API_SYSTEM_KEY
+  UKEF_TFM_API_REPORTS_KEY: UKEF_TFM_API_REPORTS_KEY
+  AZURE_NUMBER_GENERATOR_FUNCTION_SCHEDULE: AZURE_NUMBER_GENERATOR_FUNCTION_SCHEDULE
+  JWT_SIGNING_KEY: JWT_SIGNING_KEY // NOTE - in the export this appears to be a slot setting. However, we don't need to replicate that.
+}
 var tfmApiAdditionalSecureSettings = {
   UKEF_INTERNAL_NOTIFICATION: UKEF_INTERNAL_NOTIFICATION
   DTFS_CENTRAL_API_KEY: DTFS_CENTRAL_API_KEY
   EXTERNAL_API_KEY: EXTERNAL_API_KEY
   JWT_VALIDATING_KEY: JWT_VALIDATING_KEY // TODO:FN-1086 JWT_VALIDATING_KEY seems to have been moved to connection strings.
```
Comment (ChristianMcc) [inline infrastructure/resource_group_level/main.bicep:196]: and this comment too please

File: infrastructure/resource_group_level/main.bicep [hunk 3]
```diff
@@ -950,7 +947,7 @@ module frontDoorTfm 'modules/front-door-tfm.bicep' = {
 
 var tfmUiUrl = 'https://${frontDoorTfm.outputs.defaultHostName}'
 
-module tfmApiConnectionStrings 'modules/webapps/trade-finance-manager-api-connection-strings.bicep' = {
+module tfmApiConnectionStrings 'modules/webapps/trade-finance-manager-api-calculated-variables.bicep' = {
```
Comment (ChristianMcc) [inline infrastructure/resource_group_level/main.bicep:950]: i like this name

File: infrastructure/resource_group_level/modules/webapps/trade-finance-manager-api-calculated-variables.bicep [hunk 4]
```diff
@@ -36,33 +37,27 @@ resource cosmosDbAccount 'Microsoft.DocumentDB/databaseAccounts@2023-04-15' exis
...
-  AZURE_NUMBER_GENERATOR_FUNCTION_URL: {
-    type: 'Custom'
-    value: 'https://${numberGeneratorFunctionDefaultHostName}'
-  }
-  TFM_UI_URL: {
-    type: 'Custom'
-    value: tfmUiUrl
-  }
-} 
+var settingsCalculated = {
```
Comment (ChristianMcc) [inline infrastructure/resource_group_level/modules/webapps/trade-finance-manager-api-calculated-variables.bicep:40]: can this be calculatedAppSettings please just so it's the same naming as appSettings elsewhere

**Review by ChristianMcc — CHANGES_REQUESTED** [review]
Nice work! Couple of nit picks on comments/naming to remove and it's good to go

**Review by ChristianMcc — APPROVED** [review]

File: infrastructure/resource_group_level/main.bicep [hunk 5]
```diff
@@ -950,7 +947,7 @@ module frontDoorTfm 'modules/front-door-tfm.bicep' = {
...
   name: 'tfmApiConnectionStrings'
```
Comment (OxonianCambion) [inline infrastructure/resource_group_level/main.bicep:951]: The name of the module and the variable should be updated too - they are both currently `'tfmApiConnectionStrings'`

File: infrastructure/resource_group_level/modules/webapps/trade-finance-manager-api-calculated-variables.bicep [hunk 6]
```diff
@@ -1,11 +1,12 @@
-// We need to add the connection strings separately beacause the value tfmUri comes from the TFM front-door, which would produce a circular dependency.
-// See the notes in trade-finance-manager-api-no-connection-strings.bicep
+// We need to add the calculated variables separately beacause the value tfmUri comes from the TFM front-door, which would produce a circular dependency.
+// See the notes in trade-finance-manager-api-no-calculated-variables.bicep
```
Comment (OxonianCambion) [inline infrastructure/resource_group_level/modules/webapps/trade-finance-manager-api-calculated-variables.bicep:2]: You don't quite at the moment though. You set the variables earlier apart from the calculated ones, then replace them all with the combination of those and the calculated ones.

File: infrastructure/resource_group_level/modules/webapps/trade-finance-manager-api-no-calculated-variables.bicep (hunk 1 above)
Comment (OxonianCambion) [reply infrastructure/resource_group_level/modules/webapps/trade-finance-manager-api-no-calculated-variables.bicep:99]: I don't think you should set them here at all - just output them.

File: infrastructure/resource_group_level/modules/webapps/trade-finance-manager-api-no-calculated-variables.bicep [hunk 7]
```diff
@@ -1,6 +1,6 @@
-// Note that we don't set the connection strings when creating the tfm-api webapp.
+// Note that we don't set the calculated variables & connection strings when creating the tfm-api webapp.
 // This is because we need to include the TFM front door hostname which would set
-// up a circular dependency. We avoid this by setting the connection strings
+// up a circular dependency. We avoid this by setting the calculated variables
 // separately later.
```
Comment (OxonianCambion) [inline infrastructure/resource_group_level/modules/webapps/trade-finance-manager-api-no-calculated-variables.bicep:4]: Again, that's not quite what's going on. You replace all of the variables with the combination of these values and the calculated ones later. You do _not_ add the calculated ones to the existing set.

File: infrastructure/resource_group_level/modules/webapps/trade-finance-manager-api-no-calculated-variables.bicep (hunk 1 above)
Comment (francescastocco) [reply infrastructure/resource_group_level/modules/webapps/trade-finance-manager-api-no-calculated-variables.bicep:99]: Sounds good I'll update, I wasn't sure if anything needed to be set when we create the app which is why I did it this way

**Pull request merged**